nautobot/ipam/api/views.py

```diff
@@ -304,6 +305,38 @@ class IPAddressViewSet(StatusViewSetMixin, CustomFieldModelViewSet):
     serializer_class = serializers.IPAddressSerializer
     filterset_class = filters.IPAddressFilterSet
 
+    # 2.0 TODO: Remove exception class and overloaded methods below
+    # Because serializer has nat_outside as read_only, update and create methods do not need to be overloaded
+    class NatOutsideIncompatibleLegacyBehavior(APIException):
+        status_code = 412
+        default_detail = "This object does not conform to pre-1.3 behavior. Please correct data or use API version 1.3"
+        default_code = "precondition_failed"
+
+    def get_serializer_class(self):
+        if not self.brief and (
+            not hasattr(self.request, "major_version")
+            or self.request.major_version > 1
+            or (self.request.major_version == 1 and self.request.minor_version < 3)
+        ):
+            # API version 1.2 or earlier - use the legacy serializer
+            # Note: Generating API docs at this point request doesn't define major_version or minor_version for some reason
```

Comment:
Pretty sure this is expected - the "request" when generating the API docs is pretty much a dummy object only. The correct way to get the API docs correct for the different versions here would be to add `@extend_schema(version=1.2,...)` decorators to either the class or the `retrieve` and `list` methods, similar to https://github.com/nautobot/nautobot/blob/next/nautobot/extras/api/views.py#L473


I think also instead of `hasattr(self.request, "major_version")` we could also detect that we're in schema generation rather than real view execution by checking `getattr(self, 'swagger_fake_view', False)` - see https://github.com/tfranzel/drf-spectacular/commit/fffae4dc07294293decdf1e40463af47a194fa05